### DISCUSSION TITLE
Savitzky Golay Filter Java implementation there is a difference from python

### DISCUSSION THREAD

**Discussion by lixiuming1991** [psambit9791/jDSP#63] [General]
When we use Savitzky Golay Filte to smooth data,  we found that the result after Java processing is inconsistent with Python. Inconsistent intervals appear at the edges，so we checked the source code of both。When the mode selects constant, python does edge processing on the data, but Java does not.  
Excuse me, will the code be updated in the future to keep it consistent?

<img width="749" alt="image" src="https://user-images.githubusercontent.com/4647946/225581504-e0a8647a-e4f1-491f-afa4-08f237165d2c.png">
<img width="999" alt="image" src="https://user-images.githubusercontent.com/4647946/225581744-7d482461-b628-4ce5-a453-86ee574c207f.png">

**Comment by psambit9791** [comment]
@lixiuming1991 

The **interp** mode has been implemented and will be available from **version 3.0.0**. Currently available in the master branch - use JitPack to use the latest build.

To use the `interp` mode, please use the following:

```
Savgol s1 = new Savgol(7, 2);
double[] out = s1.filter(signal, "interp"); 
// default mode is "interp" - `double[] out = s1.filter(signal)` also works
```

**Comment by lixiuming1991** [comment]
Exactly !  I think using Savgol is the best compatible with our current version. Thank you for your support again !  Keep in touch！

**Comment by psambit9791** [comment]
v3.0.0 now supports `interp` mode of operation for Savgol filter (also the default mode of operation). It is now available on [Maven](https://central.sonatype.com/artifact/com.github.psambit9791/jdsp?smo=true).

**Reply by lixiuming1991** [reply]
Get it !  It will be of great help to us， thanks for your hard working .
